tls/s2n_early_data_io.c

```diff
@@ -149,6 +149,8 @@ static bool s2n_early_data_can_continue(struct s2n_connection *conn)
 S2N_RESULT s2n_send_early_data_impl(struct s2n_connection *conn, const uint8_t *data, ssize_t data_len,
         ssize_t *data_sent, s2n_blocked_status *blocked)
 {
+    RESULT_ENSURE_GTE(data_len, 0);
+    size_t validated_data_len = data_len;
```

Comment:
I would change the name of the method argument instead, since it only needs to be referenced once here. Like "data_len_in" or maybe "data_len_signed". Then your `size_t` can still be called the nicer name of "data_len".